client/scala/src/main/scala/ai/verta/blobs/dataset/AtlasHiveDatasetBlob.scala

```diff
@@ -0,0 +1,150 @@
+import ai.verta.swagger.client.HttpClient
+import ai.verta.client.entities.utils._
+
+import scala.concurrent.{Await, ExecutionContext}
+import scala.util.{Try, Success, Failure}
+import scala.concurrent.duration.Duration
+
+/** Represents a dataset from a query to an Atlas Hive table.
+ */
+class AtlasHiveDatasetBlob(
+  private val atlasQuery: String,
+  private val atlasSourceURI: String,
+  override val numRecords: Option[BigInt] = None,
+  override val executionTimestamp: Option[BigInt] = None,
+  val tags: List[String],
+  val attributes: Map[String, ValueType]
+) extends QueryDatasetBlob {
+  // cannot make this a case class due to constructor conflict.
+
+  override val query = Some(atlasQuery)
+  override val dataSourceURI = Some(atlasSourceURI)
+}
+
+object AtlasHiveDatasetBlob {
+  def apply(
+    guid: String,
+    atlasURL: String = sys.env.get("ATLAS_URL").getOrElse(""),
+    atlasUserName: String = sys.env.get("ATLAS_USERNAME").getOrElse(""),
+    atlasPassword: String = sys.env.get("ATLAS_PASSWORD").getOrElse(""),
+    atlasEntityEndpoint: String = "/api/atlas/v2/entity/bulk"
+  )(implicit ec: ExecutionContext): Try[AtlasHiveDatasetBlob] = {
+    val atlasSourceURI = f"${atlasURL}/index.html#!/detailPage/${guid}"
+    val httpClient = new HttpClient(atlasURL, Map())
+    Await.result(
+      httpClient.request(
+        "GET",
+        atlasEntityEndpoint,
+        Map(
+          "guid" -> List(guid)
+        ),
+        null,
+        jsonVal => fromJson(jsonVal, atlasSourceURI), // parser
+        Some((atlasUserName, atlasPassword)) // authentication
+      ),
+      Duration.Inf
+    ).flatten // Try[Try[AtlasHiveDatasetBlob]] to Try[AtlasHiveDatasetBlob]
+  }
+
+  private def fromJson(value: JValue, atlasSourceURI: String): Try[AtlasHiveDatasetBlob] =
+    value match {
+      case JObject(fields) => {
+        val fieldsMap = fields.map(f => (f.name, f.value)).toMap
+
+        for (
+          entityMap <- extractEntity(fieldsMap);
+          _ <- checkEntityType(entityMap); // return Failure right away if type is not hive table
+          attributesMap <- getSubMap(entityMap, "attributes");
+          relationshipAttributesMap <- getSubMap(entityMap, "relationshipAttributes");
+          dbRelationshipAttributesMap <- getSubMap(relationshipAttributesMap, "db");
+          parametersMap <- getSubMap(attributesMap, "parameters")
+        ) yield {
+          val tags = getTags(entityMap)
+          val attributes = getAttributes(entityMap, attributesMap, dbRelationshipAttributesMap)
+
+          val tableName: String = attributes.get("table_name").get.asString.get
+          val databaseName: String = attributes.get("database_name").get.asString.get
+          val atlasQuery = f"select * from ${databaseName}.${tableName}"
+
+          val numRecords: BigInt = parametersMap.get("numRows").map(JsonConverter.fromJsonInteger).get
+
+          // this is based on the Python client, but is it correct?
```

Comment:
I think MDB saves timestamp as milliseconds.

Reply:
I am pretty sure that this timestamp is in milliseconds (since 1970 or sth like that, 13 digits).
Update: yep, can confirm. It returns this: 1600206916216

Reply:
I am pretty sure that this timestamp is in milliseconds (since 1970 or sth like that, 13 digits).
Update: yep, can confirm. It returns this: 1600206916216